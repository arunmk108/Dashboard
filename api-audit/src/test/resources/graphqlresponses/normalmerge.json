{
    "data": {
        "repository": {
            "pullRequests": {
                "totalCount": 1,
                "pageInfo": {
                    "endCursor": "Y3Vyc29yOnYyOpK5MjAxNy0xMi0xMlQxMzowOTozOS0wNTowMM4ADTnZ",
                    "hasNextPage": false
                },
                "edges": [
                    {
                        "cursor": "Y3Vyc29yOnYyOpK5MjAxNy0xMi0xMlQxMzowOTozOS0wNTowMM4ADTnZ",
                        "node": {
                            "number": 1,
                            "mergeable": "UNKNOWN",
                            "state": "MERGED",
                            "createdAt": "2017-12-12T18:09:27Z",
                            "updatedAt": "2017-12-12T18:09:39Z",
                            "mergedAt": "2017-12-12T18:09:39Z",
                            "baseRef": {
                                "target": {
                                    "oid": "ee72b85052566d33a83dd57f22137ddcd9285dfd"
                                }
                            },
                            "baseRefName": "master",
                            "title": "pr from fork",
                            "mergeCommit": {
                                "oid": "6a5c56e4f5c9e72702af77c4d69030f44188a80b"
                            },
                            "headRepository": {
                                "name": "NormalMerge",
                                "url": "https://mygithub.com/ghtwo/NormalMerge"
                            },
                            "headRef": {
                                "target": {
                                    "oid": "204a0983489cf263e0cf6e058e630a74498f9e34"
                                }
                            },
                            "headRefName": "master",
                            "author": {
                                "login": "ghtwo",
                                "resourcePath": "/ghtwo"
                            },
                            "commits": {
                                "totalCount": 2,
                                "nodes": [
                                    {
                                        "commit": {
                                            "oid": "613330d0d6482f9e56d09051ff4cd082238c8f21",
                                            "parents": {
                                                "nodes": [
                                                    {
                                                        "oid": "1ca6c08c7435be134b51f44325527667982afee1"
                                                    }
                                                ]
                                            },
                                            "committedDate": "2017-12-12T18:08:39Z",
                                            "message": "Third fork commit",
                                            "status": null,
                                            "author": {
                                                "name": "Author, Two",
                                                "date": "2017-12-12T13:08:39-05:00",
                                                "user": {
                                                    "login": "ghtwo"
                                                }
                                            }
                                        }
                                    },
                                    {
                                        "commit": {
                                            "oid": "204a0983489cf263e0cf6e058e630a74498f9e34",
                                            "parents": {
                                                "nodes": [
                                                    {
                                                        "oid": "613330d0d6482f9e56d09051ff4cd082238c8f21"
                                                    }
                                                ]
                                            },
                                            "committedDate": "2017-12-12T18:08:55Z",
                                            "message": "Fourth fork commit",
                                            "status": null,
                                            "author": {
                                                "name": "Author, Two",
                                                "date": "2017-12-12T13:08:55-05:00",
                                                "user": {
                                                    "login": "ghtwo"
                                                }
                                            }
                                        }
                                    }
                                ]
                            },
                            "comments": {
                                "totalCount": 0,
                                "nodes": []
                            },
                            "reviews": {
                                "totalCount": 0,
                                "nodes": []
                            }
                        }
                    }
                ]
            },
            "ref": {
                "target": {
                    "history": {
                        "pageInfo": {
                            "endCursor": "ee72b85052566d33a83dd57f22137ddcd9285dfd 5",
                            "hasNextPage": false
                        },
                        "edges": [
                            {
                                "cursor": "ee72b85052566d33a83dd57f22137ddcd9285dfd 0",
                                "node": {
                                    "oid": "ee72b85052566d33a83dd57f22137ddcd9285dfd",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "6a5c56e4f5c9e72702af77c4d69030f44188a80b"
                                            }
                                        ]
                                    },
                                    "message": "Sixth direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:10:44-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:10:44-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "ee72b85052566d33a83dd57f22137ddcd9285dfd 1",
                                "node": {
                                    "oid": "6a5c56e4f5c9e72702af77c4d69030f44188a80b",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "1ca6c08c7435be134b51f44325527667982afee1"
                                            },
                                            {
                                                "oid": "204a0983489cf263e0cf6e058e630a74498f9e34"
                                            }
                                        ]
                                    },
                                    "message": "Merge pull request #1 from ghtwo/master\n\npr from fork",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:09:37-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:09:37-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "ee72b85052566d33a83dd57f22137ddcd9285dfd 2",
                                "node": {
                                    "oid": "204a0983489cf263e0cf6e058e630a74498f9e34",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "613330d0d6482f9e56d09051ff4cd082238c8f21"
                                            }
                                        ]
                                    },
                                    "message": "Fourth fork commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:08:55-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:08:55-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "ee72b85052566d33a83dd57f22137ddcd9285dfd 3",
                                "node": {
                                    "oid": "613330d0d6482f9e56d09051ff4cd082238c8f21",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "1ca6c08c7435be134b51f44325527667982afee1"
                                            }
                                        ]
                                    },
                                    "message": "Third fork commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:08:39-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:08:39-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "ee72b85052566d33a83dd57f22137ddcd9285dfd 4",
                                "node": {
                                    "oid": "1ca6c08c7435be134b51f44325527667982afee1",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "8b89177aa00d7c8cba7adf8d46bea51964380b7c"
                                            }
                                        ]
                                    },
                                    "message": "Second direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:08:05-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:08:05-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "ee72b85052566d33a83dd57f22137ddcd9285dfd 5",
                                "node": {
                                    "oid": "8b89177aa00d7c8cba7adf8d46bea51964380b7c",
                                    "parents": {
                                        "nodes": []
                                    },
                                    "message": "First direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:07:49-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:07:49-05:00"
                                    },
                                    "status": null
                                }
                            }
                        ]
                    }
                }
            },
            "issues": {
                "totalCount": 0,
                "pageInfo": {
                    "endCursor": null,
                    "hasNextPage": false
                },
                "edges": []
            }
        }
    }
}