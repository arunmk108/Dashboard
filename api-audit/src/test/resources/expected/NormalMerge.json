[
    {
        "auditStatuses": [
            "GIT_FORK_STRATEGY",
            "COMMITAUTHOR_EQ_MERGECOMMITER",
            "PULLREQ_NOT_PEER_REVIEWED"
        ],
        "lastUpdated": 1516207279841,
        "pullRequest": {
            "scmUrl": "https://mygithub.com/Devopscode/NormalMerge",
            "scmBranch": "master",
            "scmRevisionNumber": "6a5c56e4f5c9e72702af77c4d69030f44188a80b",
            "scmCommitLog": "pr from fork",
            "scmCommitTimestamp": 1513102179000,
            "numberOfChanges": 0,
            "id": "5a5f77a16fa3ba5285b97539",
            "orgName": "Devopscode",
            "repoName": "NormalMerge",
            "sourceRepo": "ghtwo/NormalMerge",
            "sourceBranch": "master",
            "targetRepo": "Devopscode/NormalMerge",
            "targetBranch": "master",
            "number": "1",
            "collectorItemId": "5a4e549e33ec296b81c3938b",
            "updatedAt": 1513102179000,
            "createdAt": 1513102167000,
            "closedAt": 0,
            "state": "merged",
            "mergedAt": 1513102179000,
            "timestamp": 1513102167000,
            "resolutiontime": 0,
            "userId": "ghtwo",
            "comments": [],
            "reviews": [],
            "headSha": "204a0983489cf263e0cf6e058e630a74498f9e34",
            "baseSha": "ee72b85052566d33a83dd57f22137ddcd9285dfd",
            "requestType": "pull",
            "commits": [
                {
                    "scmRevisionNumber": "613330d0d6482f9e56d09051ff4cd082238c8f21",
                    "scmCommitLog": "Third fork commit",
                    "scmAuthor": "Author, Two",
                    "scmAuthorLogin": "ghtwo",
                    "scmParentRevisionNumbers": [
                        "1ca6c08c7435be134b51f44325527667982afee1"
                    ],
                    "scmCommitTimestamp": 1513102119000,
                    "numberOfChanges": 0,
                    "timestamp": 0,
                    "firstEverCommit": false
                },
                {
                    "scmRevisionNumber": "204a0983489cf263e0cf6e058e630a74498f9e34",
                    "scmCommitLog": "Fourth fork commit",
                    "scmAuthor": "Author, Two",
                    "scmAuthorLogin": "ghtwo",
                    "scmParentRevisionNumbers": [
                        "613330d0d6482f9e56d09051ff4cd082238c8f21"
                    ],
                    "scmCommitTimestamp": 1513102135000,
                    "numberOfChanges": 0,
                    "timestamp": 0,
                    "firstEverCommit": false
                }
            ]
        },
        "commits": [
            {
                "scmRevisionNumber": "613330d0d6482f9e56d09051ff4cd082238c8f21",
                "scmCommitLog": "Third fork commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "1ca6c08c7435be134b51f44325527667982afee1"
                ],
                "scmCommitTimestamp": 1513102119000,
                "numberOfChanges": 0,
                "timestamp": 0,
                "firstEverCommit": false
            },
            {
                "scmRevisionNumber": "204a0983489cf263e0cf6e058e630a74498f9e34",
                "scmCommitLog": "Fourth fork commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "613330d0d6482f9e56d09051ff4cd082238c8f21"
                ],
                "scmCommitTimestamp": 1513102135000,
                "numberOfChanges": 0,
                "timestamp": 0,
                "firstEverCommit": false
            }
        ],
        "scmUrl": "https://mygithub.com/Devopscode/NormalMerge",
        "scmBranch": "master"
    },
    {
        "auditStatuses": [
            "SCM_AUTHOR_LOGIN_INVALID",
            "DIRECT_COMMITS_TO_BASE_FIRST_COMMIT",
            "DIRECT_COMMITS_TO_BASE"
        ],
        "lastUpdated": 1516207279841,
        "commits": [
            {
                "scmUrl": "https://mygithub.com/Devopscode/NormalMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "ee72b85052566d33a83dd57f22137ddcd9285dfd",
                "scmCommitLog": "Sixth direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "6a5c56e4f5c9e72702af77c4d69030f44188a80b"
                ],
                "scmCommitTimestamp": 1513102244000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a06fa3ba5285b97533",
                "collectorItemId": "5a4e549e33ec296b81c3938b",
                "timestamp": 1516205984995,
                "firstEverCommit": false
            },
            {
                "scmUrl": "https://mygithub.com/Devopscode/NormalMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "1ca6c08c7435be134b51f44325527667982afee1",
                "scmCommitLog": "Second direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "8b89177aa00d7c8cba7adf8d46bea51964380b7c"
                ],
                "scmCommitTimestamp": 1513102085000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a06fa3ba5285b97537",
                "collectorItemId": "5a4e549e33ec296b81c3938b",
                "timestamp": 1516205984995,
                "firstEverCommit": false
            },
            {
                "scmUrl": "https://mygithub.com/Devopscode/NormalMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "8b89177aa00d7c8cba7adf8d46bea51964380b7c",
                "scmCommitLog": "First direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [],
                "scmCommitTimestamp": 1513102069000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a06fa3ba5285b97538",
                "collectorItemId": "5a4e549e33ec296b81c3938b",
                "timestamp": 1516205984995,
                "firstEverCommit": true
            }
        ],
        "scmUrl": "https://mygithub.com/Devopscode/NormalMerge",
        "scmBranch": "master"
    }
]